.github/workflows/scripts/util.js

```diff
@@ -0,0 +1,150 @@
+// @ts-check
```

Comment:
Do these scripts need to live under the workflows directory?

Reply:
Unless there is a strong reason I would like to keep these under eng/scripts with the other shared scripts.

Reply:
They can live anywhere in the repo, but I like nesting them under `.github/workflows` for a few reasons:

1. You can sparse clone `.github/workflows` to get the YAML and the supporting scripts
2. These scripts are different than the scripts under `eng`, in that they assume they are running in the `github-script` context (and only work in such a context), rather than being run as a standard console app.  They are also implemented in JS rather than TS.

So I'd rather not mix these directly in with the existing scripts under `eng/scripts`, but I'm fine putting them **somewhere** under `eng` if you strongly prefer it over `.github/workflows`.